ecssd_agent.go

```diff
@@ -222,24 +232,28 @@ func deleteDNSRecord(serviceName string, dockerId string) error {
 	if err != nil {
 		return err
 	}
-	aValue := ""
-	for _, rrset := range resp.ResourceRecordSets {
-		if *rrset.SetIdentifier == dockerId {
-			for _, rrecords := range rrset.ResourceRecords {
-				aValue = aws.StringValue(rrecords.Value)
-				break
-			}
-			// Break out of outer loop early if we found the record.
-			if aValue != "" {
-				break
-			}
-		}
-	}
-	if aValue == "" {
+	// Purge the A records of the local IP
+	aValues := resp.ResourceRecordSets[0].ResourceRecords
+	if aValues == nil {
 		log.Error("Route53 Record doesn't exist")
 		return nil
 	}
 
+	changedRecord := false
+
+	for i, aValue := range aValues {
+		if *aValue.Value == configuration.LocalIp {
+			// delete this ResourceRecord
+			aValues = append(aValues[:i], aValues[i+1])
```

Comment:
Should be 

`aValues = append(aValues[:i], ...aValues[i+1:])`

Could also write like `N = len(aValues) - 1; aValues[i] = aValues[N - 1]; aValues = aValues[:N-1];`, but that's less clear.